torch/lib/THC/THCTensorScatterGather.cu

```diff
@@ -92,7 +92,7 @@ __global__ void THCudaTensor_gatherKernel(
                                                           tensor, &tensorOffset,
                                                           src, &srcOffset);
 
-    IndexType indexValue = (IndexType)index.data[indexOffset] - TH_INDEX_BASE;
+    long indexValue = index.data[indexOffset] - TH_INDEX_BASE;
```

Comment:
Wouldn't it be better to use a `long long` here? Just to be safe -- `ULONG_MAX > LONG_MAX`